src/covid_shared/cli_tools/decorators.py

```diff
@@ -48,3 +50,109 @@ def _wrapped(*args, **kwargs):
         return _wrapped
 
     return _pass_run_metadata
+
+
+def add_r_singularity_option():
+    """
+    Decorator to set R environment to use in a CLI command.
+    """
+    def wrapper(entry_point: types.FunctionType):
+        entry_point = click.option(
+            '--r-singularity-image', 'r_singularity_image',
+            type=click.Path(file_okay=True),
+            default=R_SINGULARITY_IMAGE_PATH,
+            help=(f"Path to a specific R singularity. Can pass 'None' to use active conda environment. "
+                  f"By default uses {R_SINGULARITY_IMAGE_PATH}.")
+        )(entry_point)
+
+        return entry_point
+    return wrapper
+
+
+def add_create_directory_options(default_output_root: Path):
+    """
+    Decorator to set CLI command options required when producing outputs.
+    """
+    def wrapper(entry_point: types.FunctionType):
+
+        entry_point = click.option(
+            "-o", "--output-root",
+            type=click.Path(file_okay=False),
+            default=default_output_root,
+            help=(f"Directory to outputs. Defaults to {default_output_root}/YYYY_MM_DD.VV for today's date and the "
+                  f"newest uncreated version")
+        )(entry_point)
+
+        entry_point = click.option(
+            '-p', '--production-tag',
+            type=click.STRING,
+            help='Tags this run as a production run.'
+        )(entry_point)
+
+        entry_point = click.option(
+            '-b', '--mark-best',
+            is_flag=True,
+            default=False,
+            help='Mark this run as "best"'
+        )(entry_point)
+
+        return entry_point
+    return wrapper
+
+
+def _create_dependency_option(data_source: str, default: Path = BEST_LINK):
+    """
+    Declares a dependency on a specific version of some data source, with a default option
+
+    Parameters
+    ----------
+    data_source
+        The name of the data source. Multiple words should be separated by '-'
+    default
+        The default location of the dependency. If None is provided, there is no default and the option is required.
+    """
+    def wrapper(entry_point: types.FunctionType):
+        click_args = {
+            'type': click.Path(file_okay=False),
+            'help': f'Version of the {data_source.replace("-", " ")} to use. Defaults to "{default}"',
```

Comment:
`help` text incomplete - if default is None there actually is no default and it's mandatory. Maybe set `help` situationally in the below if/else block?

Reply:
Good call.